keras_nlp/tokenizers/byte_pair_tokenizer.py

```diff
@@ -100,12 +101,30 @@ def split_strings_for_bpe(inputs):
     inputs = tf.strings.regex_replace(
         inputs, rf"(\s{SPECIAL_WHITESPACES})$", r"\1६"
     )
+    if special_tokens:
+
+        def build_alt(i):
+            return " " + "I" * i + "IGuessIamAValidWord" + "d" * i
```

Comment:
heh this feels a little too janky :)

What if we...

 - Move this function out (don't nest it) for readability.
 - Take in the original special_token.
 - Strip all splittable `\s\p{L}\p{N}` from the original special token.
 - Add a unique prefix or suffix. `f"Ĵ{i}"` or something like that.

That is nice in that it won't expand the length of these characters much, and if someone is debugging intermediate state, it will look a little more readable.

Reply:
haha done.